src/Microsoft.SqlTools.ServiceLayer/Cms/CmsService.cs

```diff
@@ -0,0 +1,434 @@
+                {
+                    serverConn = ConnectionService.OpenServerConnection(connInfo);
+                }
+            }
+            return serverConn;
+        }
+
+        private ListRegisteredServersResult GetChildrenfromParentGroup(ServerGroup parentGroup)
+        {
+            var servers = parentGroup.RegisteredServers;
+            var groups = parentGroup.ServerGroups;
+
+            // Convert to appropriate variables and return
+            var serverResults = new List<RegisteredServerResult>();
+            foreach (var s in servers)
+            {
+                serverResults.Add(new RegisteredServerResult
+                {
+                    Name = s.Name,
+                    ServerName = s.ServerName,
+                    Description = s.Description,
+                    connectionDetails = ConnectionServiceInstance.ParseConnectionString(s.ConnectionString),
+                    RelativePath = s.KeyChain.Urn.SafeToString()
+                });
+            }
+
+            var groupsResults = new List<RegisteredServerGroup>();
+            foreach (var s in groups)
+            {
+                groupsResults.Add(new RegisteredServerGroup
+                {
+                    Name = s.Name,
+                    Description = s.Description,
+                    RelativePath = s.KeyChain.Urn.SafeToString()
+                });
+            }
+            ListRegisteredServersResult result = new ListRegisteredServersResult() { RegisteredServersList = serverResults, RegisteredServerGroups = groupsResults };
+            return result;
+        }
+
+        #endregion
+
+        /// <summary>
+        /// Internal for testing purposes only
+        /// </summary>
+        internal static ConnectionService ConnectionServiceInstance
+        {
+            get
+            {
+                if (connectionService == null)
+                {
+                    connectionService = ConnectionService.Instance;
+                }
+                return connectionService;
+            }
+            set
+            {
+                connectionService = value;
+            }
+        }
+
+        /// <summary>
+        /// Internal for testing purposes only
+        /// </summary>
+        internal static RegisteredServersStore ServerStore
+        {
+            get
+            {
+                if (registerdServerStore == null)
+                {
+                    registerdServerStore = RegisteredServersStore.LocalFileStore;
+                }
+                return registerdServerStore;
+            }
+            set
+            {
+                lock (lockObject)
+                {
+                    registerdServerStore = value;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Internal variable for testability
+        /// </summary>
+        internal Task CmsTask { get; set; }
```

Comment:
what are you using this property for?  I don't see it being accessed outside this class, and then only to assign to it.

Reply:
This is only used in tests. Since the call to Handle functions in Service is async, we need a way to wait for task to complete to validate requestContext result. This was one pattern used for testing across multiple services so used the same.

Reply:
makes sense.  For some reason when I search for usages yesterday I didn't see any, but now I see this being used through out the tests.  Maybe I had a typo in search box 😄